yunwoo/programmers-2/귤-고르기.js

```diff
@@ -0,0 +1,30 @@
+function solution(k, tangerine) {
+  const tangerineMapObject = {};
+  let result = 1;
+  let count = 0;
+
+  tangerine.forEach(
+    (el) => (tangerineMapObject[el] = (tangerineMapObject[el] || 0) + 1)
+  );
+
+  const tangerineQuantitys = Object.values(tangerineMapObject).sort(
+    (a, b) => b - a
+  );
+
+  for (const el of tangerineQuantitys) {
+    if (count + el < k) {
+      count += el;
+      result++;
+    } else {
+      return result;
+    }
+  }
+}
```

Comment:
```suggestion
function solution(k, tangerine) {
  const tangerineMapObject = {};
  let result = 1;
  let count = 0;

  tangerine.forEach(
    (el) => (tangerineMapObject[el] = (tangerineMapObject[el] || 0) + 1)
  );

  const tangerineQuantitys = Object.values(tangerineMapObject).sort(
    (a, b) => b - a
  );

  for (const el of tangerineQuantitys) {
    if (count + el >= k) return result;
    count += el;
    result++;
  }
}

```
- 풀이잘봤습니다 리턴 조건식을 먼저 작성함으로써 else 구문을 제거할 수 있습니다! 🙇‍♂️
- 해당 문제는 객체가 아닌 배열로서 풀면 Object.values와 같은 추가적인 작업이 필요없어집니다
```js
function solution(k, tangerine) {
  const tangerineCountArr = tangerine.reduce(
    (acc, cur) => {
      acc[cur] = (acc[cur] || 0) + 1;
      return acc;
    },
    [0]
  );

  let sum = 0;
  let result = 0;

  tangerineCountArr.sort((a, b) => b - a);

  for (const count of tangerineCountArr) {
    sum += count;
    result++;

    if (sum >= k) return result;
  }
}
```

Reply:
- 훨씬깔끔하네요!! 배열로 접근하는것도 고려해보도록 노력하겠습니다 감사합니다 👍